langchain4j-core/src/main/java/dev/langchain4j/store/embedding/EmbeddingStore.java

```diff
@@ -52,14 +54,39 @@ public interface EmbeddingStore<Embedded> {
      */
     List<String> addAll(List<Embedding> embeddings);
 
+    /**
+     * Adds multiple embeddings and their corresponding contents that have been embedded to the store.
+     *
+     * @param ids A list of embeddings ids associated with the added embeddings.
+     * @param embeddings A list of embeddings to be added to the store.
+     * @param embedded   A list of original contents that were embedded.
+     * @return A list of IDs associated with the added embeddings.
+     */
+    List<String> addAll(List<String> ids, List<Embedding> embeddings, List<Embedded> embedded);
+
     /**
      * Adds multiple embeddings and their corresponding contents that have been embedded to the store.
      *
      * @param embeddings A list of embeddings to be added to the store.
      * @param embedded   A list of original contents that were embedded.
      * @return A list of auto-generated IDs associated with the added embeddings.
      */
-    List<String> addAll(List<Embedding> embeddings, List<Embedded> embedded);
+    default List<String> addAll(List<Embedding> embeddings, List<Embedded> embedded) {
+        return addAll(generateIds(embeddings.size()), embeddings, embedded);
+    }
+
+
+    /**
+     * Generates list of UUID strings
+     * @param dimension  - dimension of list
+     */
+    default List<String> generateIds(long dimension) {
```

Comment:
```suggestion
    default List<String> generateIds(int n) {
```
`dimension` has a specific meaning in the context of embeddings, so let's better rename it into `n`

Reply:
+++